bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/communication/RemoteAPIController.java

```diff
@@ -0,0 +1,149 @@
+                    response.getContentAsString(), contentType);
+
+            if (HttpStatus.isSuccess(response.getStatus())) {
+                String stringResponse = response.getContentAsString().trim();
+
+                if (null != contentType)
+                    switch (contentType) {
+                        case CONTENTTYPE_JSON:
+                            return JsonParserHelper.parseJson(gson, stringResponse);
+                        case CONTENTTYPE_TEXT:
+                            return PrometheusParserHelper.parsePrometheus(stringResponse);
+                        case CONTENTTYPE_OPENMETRICS:
+                            return PrometheusParserHelper.parsePrometheus(stringResponse);
+                        default:
+                            logger.debug("Unhandled content type returned: {}", contentType);
+                    }
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    public void setLedMode(String serialNo, String mode) {
+        Request request = httpClient.newRequest(RESTHelper.generateGetLedsModeUrl(apiConfig, serialNo));
+        request.timeout(REQUEST_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        request.method(HttpMethod.PUT);
+        request.header(HttpHeader.CONTENT_TYPE, CONTENTTYPE_JSON);
+        LedMode ledMode = new LedMode();
+        ledMode.mode = mode;
+        String modeJson = gson.toJson(ledMode);
+        logger.debug("Setting LEDS mode for {}: {}", serialNo, modeJson);
+        request.content(new StringContentProvider(CONTENTTYPE_JSON, modeJson, StandardCharsets.UTF_8));
+        sendRequest(request);
+    }
+
+    public void calibrateCo2(String serialNo) {
+        logger.debug("Triggering CO2 calibration for {}", serialNo);
+        sendRequest(RESTHelper.generateRequest(httpClient, RESTHelper.generateCalibrationCo2Url(apiConfig, serialNo),
+                HttpMethod.POST));
+    }
+
+    private @Nullable ContentResponse sendRequest(@Nullable final Request request) {
+        if (request == null) {
+            handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Unable to generate request");
+        }
+
+        @Nullable
+        ContentResponse response = null;
+        try {
+            response = request.send();
+            if (response != null) {
+                logger.debug("Response from {}: {}", request.getURI(), response.getStatus());
+                if (HttpStatus.isSuccess(response.getStatus())) {
+                    handler.updateStatus(ThingStatus.ONLINE);
+                } else {
+                    handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Returned status code: " + response.getStatus());
+                }
+            } else {
+                handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "No response");
```

Comment:
```suggestion
                    throw new ApiGradientNoResponseException();
```

This custom excpetion should be created and can derive from `Exception`